commons/com.b2international.index/src/com/b2international/index/es/query/Es8QueryBuilder.java

```diff
@@ -0,0 +1,499 @@
+			// XXX executing a term query on a probably nonexistent field and value, should return zero docs
+			deque.push(MATCH_NONE);
+		} else if (expression instanceof StringPredicate) {
+			visit((StringPredicate) expression);
+		} else if (expression instanceof LongPredicate) {
+			visit((LongPredicate) expression);
+		} else if (expression instanceof LongRangePredicate) {
+			visit((LongRangePredicate) expression);
+		} else if (expression instanceof StringRangePredicate) {
+			visit((StringRangePredicate) expression);
+		} else if (expression instanceof NestedPredicate) {
+			visit((NestedPredicate) expression);
+		} else if (expression instanceof HasParentPredicate) {
+			visit((HasParentPredicate) expression);
+		} else if (expression instanceof PrefixPredicate) {
+			visit((PrefixPredicate) expression);
+		} else if (expression instanceof RegexpPredicate) {
+			visit((RegexpPredicate) expression);
+		} else if (expression instanceof StringSetPredicate) {
+			visit((StringSetPredicate) expression);
+		} else if (expression instanceof LongSetPredicate) {
+			visit((LongSetPredicate) expression);
+		} else if (expression instanceof IntPredicate) {
+			visit((IntPredicate) expression);
+		} else if (expression instanceof IntSetPredicate) {
+			visit((IntSetPredicate) expression);
+		} else if (expression instanceof BoolExpression) {
+			visit((BoolExpression) expression);
+		} else if (expression instanceof BooleanPredicate) {
+			visit((BooleanPredicate) expression);
+		} else if (expression instanceof IntRangePredicate) {
+			visit((IntRangePredicate) expression);
+		} else if (expression instanceof TextPredicate) {
+			visit((TextPredicate) expression);
+		} else if (expression instanceof DisMaxPredicate) {
+			visit((DisMaxPredicate) expression);
+		} else if (expression instanceof BoostPredicate) {
+			visit((BoostPredicate) expression);
+		} else if (expression instanceof ScriptScoreExpression) {
+			visit((ScriptScoreExpression) expression);
+		} else if (expression instanceof DecimalPredicate) {
+			visit((DecimalPredicate) expression);
+		} else if (expression instanceof DecimalRangePredicate) {
+			visit((DecimalRangePredicate) expression);
+		} else if (expression instanceof DecimalSetPredicate) {
+			visit((DecimalSetPredicate) expression);
+		} else if (expression instanceof DoublePredicate) {
+			visit((DoublePredicate) expression);
+		} else if (expression instanceof DoubleRangePredicate) {
+			visit((DoubleRangePredicate) expression);
+		} else if (expression instanceof DoubleSetPredicate) {
+			visit((DoubleSetPredicate) expression);
+		} else if (expression instanceof ScriptQueryExpression){
+			visit((ScriptQueryExpression)expression);
+		} else {
+			throw new IllegalArgumentException("Unexpected expression: " + expression);
+		}
+	}
+
+	private void visit(ScriptQueryExpression expression) {
+		throw new UnsupportedOperationException();
+//		deque.push(QueryBuilders.scriptQuery(expression.toEsScript(mapping)));
```

Comment:
Please add a ticket in Jira for all the commented sections, so we know what gaps we need to fill in later.

Reply:
Implemented all remaining expression to query transformations.